ns1/datadog_checks/ns1/check.py

```diff
@@ -0,0 +1,448 @@
+import errno
+import json
+import logging
+import os
+
+from requests.exceptions import ConnectionError, HTTPError, InvalidURL, Timeout
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .ns1_url_utils import Ns1Url
+
+
+class Ns1Check(AgentCheck):
+    NS1_SERVICE_CHECK = "ns1.can_connect"
+
+    def __init__(self, name, init_config, instances):
+        super(Ns1Check, self).__init__(name, init_config, instances)
+        self.api_endpoint = self.instance.get("api_endpoint")
+        if not self.api_endpoint:
+            raise ConfigurationError('NS1 API endpoint must be specified in configuration')
+
+        self.api_key = self.instance.get("api_key")
+        if not self.api_key:
+            raise ConfigurationError('NS1 API key must be specified in configuration')
+
+        self.headers = {"X-NSONE-Key": self.api_key}
+
+        self.metrics = self.instance.get("metrics")
+        if not self.metrics or len(self.metrics) == 0:
+            raise ConfigurationError('Invalid metrics config!')
+
+        self.query_params = self.instance.get("query_params")
+        self.usage_count_path = "/opt/datadog-agent/log"
+        self.usage_count_fname = 'ns1_usage_count.txt'
+        self.service_check(
+            self.NS1_SERVICE_CHECK,
+            AgentCheck.OK,
+            message='Configuration check for NS1 API endpoint %s was successful' % self.api_endpoint,
+        )
+        self.ns1 = Ns1Url(self.api_endpoint)
+        self.pulsar_apps = {}
+
+    def check(self, instance):
+
+        self.set_logger('/opt/datadog-agent/log/ns1.log')
+        logging.info('Startup')
+
+        # get counters from previous run
+        self.get_usage_count()
+
+        # create URLs to query API for all configured metrics
+        checkUrl = self.create_url(self.metrics, self.query_params)
+
+        for k, v in checkUrl.items():
+            try:
+                url, name, tags, metric_type = v
+                # Query API to get metrics
+                res = self.get_stats(url)
+                logging.info('NS1 API result', extra={'result': res})
+                logging.info(json.dumps(res))
+                if res:
+                    # extract metric from API result.
+                    val, status = self.extract_metric(k, res)
+
+                    # send metric to datadog if extraction was sucessful
+                    if status:
+                        self.send_metrics(name, val, tags, metric_type)
+
+            except Exception:
+                raise
+        # save counters for next run
+        self.set_usage_count()
+
+    def set_logger(self, logfile):
+        logging.basicConfig(
```

Comment:
Same story, logging to a separate file for a check is anti-pattern. Use `self.log` to log to the agent log file in a standard way.